plugins/all-modules-page/src/main/kotlin/ExternalModuleLinkResolver.kt

```diff
@@ -52,15 +51,19 @@ class DefaultExternalModuleLinkResolver(val context: DokkaContext) : ExternalMod
         }
 
     override fun resolve(dri: DRI, fileContext: File): String? {
-        val absoluteLink = elps.mapNotNull { it.resolve(dri) }.firstOrNull() ?: return null
-        val modulePath = context.configuration.outputDir.absolutePath.split(File.separator)
-        val contextPath = fileContext.absolutePath.split(File.separator)
-        val commonPathElements = modulePath.zip(contextPath)
+        val resolvedLinks = elps.mapNotNull { it.resolve(dri)?.removePrefix("file:") }
+        val modulePath = context.configuration.outputDir.absolutePath
+        val validLink = resolvedLinks.firstOrNull {
+            val absolutePath = File(modulePath + it)
+            absolutePath.isFile
```

Comment:
This checks assumes that dependee submodules have be already processed but that is upto process order  
in `all-modules-page` plugin. For our project it lead to situation where same DRI was sometimes resolved and sometimes not.
I [workaround](https://github.com/pbajurko/dokka/commit/b420868da055ae9fd0e6f583ebbaab6a4eafcbc4) it by checking paths against Dokka's output in submodules since partial tasks will be done before multimodule starts but that is not clean solution.

Reply:
I agree. Thank you for noticing it. The [workaround](https://github.com/pbajurko/dokka/commit/b420868da055ae9fd0e6f583ebbaab6a4eafcbc4) is correct, but it should not depend on particular paths like `build/dokka/htmlMultiModule` or `build/dokka/htmlPartial/`. It would be great to use [`module.sourceOutputDirectory`](https://github.com/Kotlin/dokka/blob/8e5c63d035ef44a269b8c43430f43f5c8eebfb63/dokka-subprojects/core/src/main/kotlin/org/jetbrains/dokka/configuration.kt#L240)